OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -98,22 +98,26 @@ class ProximityTrigger : IDisposable
 
 			WPos position;
 			WDist range;
+			WDist vRange;
+
 			IEnumerable<Actor> currentActors = Enumerable.Empty<Actor>();
 
-			public ProximityTrigger(WPos pos, WDist range, Action<Actor> onActorEntered, Action<Actor> onActorExited)
+			public ProximityTrigger(WPos pos, WDist range, WDist vRange, Action<Actor> onActorEntered, Action<Actor> onActorExited)
 			{
 				this.onActorEntered = onActorEntered;
 				this.onActorExited = onActorExited;
 
-				Update(pos, range);
+				Update(pos, range, vRange);
 			}
 
-			public void Update(WPos newPos, WDist newRange)
+			public void Update(WPos newPos, WDist newRange, WDist newVRange)
 			{
 				position = newPos;
 				range = newRange;
+				vRange = newVRange;
+
+				var offset = new WVec(newRange, newRange, newVRange.Length != 0 ? newVRange : WDist.Zero);
```

Comment:
If the length is 0, isn't `newVRange` not already the same as `WDist.Zero`?
So this can be simplified to `var offset = new WVec(newRange, newRange, newVRange);`